app/scripts/models/user.js

```diff
@@ -372,7 +373,10 @@ define(function (require, exports, module) {
       var self = this;
       return account.signIn(password, relier, options)
         .then(function () {
-          if (! account.get('verified')) {
+          var isSignUp =
```

Comment:
We *are* already doing this, just check whether `verificationReason` is `SIGN_UP` here instead.